### ISSUE TITLE
"Network Error" when using self-signed certificate 

### ISSUE THREAD

**Issue by mrjovanovic** [haiwen/seadroid#116]
Our collective is trying out Seafile on a self-hosted server with a self-signed SSL certificate. Even though the certificate is correctly imported into the android system, Seadroid still shows "Network Error" when trying to log in with my credentials. 

Does Seadroid use the android system certificates? The proper behavior should be that, like other android apps, Seadroid would use system certificates when available. If Seadroid uses its own certificate repository, there should be an option to import certificates to the app.

As of now, none of us can use the app on our Android devices.

[Screenshot of error](https://cloud.githubusercontent.com/assets/4664783/4461353/aa02870c-48b9-11e4-9ff3-6ec05badfdde.png). I'd be happy to debug, if someone could tell me where to look!

EDIT:
Another thing that might help: curl -d "username=username&password=password" https://website.something/seafileapi2/auth-token gives this error:

```
 curl: (60) SSL certificate problem: unable to get local issuer certificate
 More details here: http://curl.haxx.se/docs/sslcerts.html
```

when curl is used with the option --insecure (meant to ignore that the cert is self-signed), it shoots out an HTML page that starts off like this:

``` html
    <!DOCTYPE html>
    <html xmlns="http://www.w3.org/1999/xhtml">
    <head>
    <title>Private Seafile</title>
    <meta http-equiv="Content-type" content="text/html; charset=utf-8" />
    <meta name="keywords" content="File Collaboration Team Organization" /> 
    ...
```

The curl command was run on a Linux box, which has the certificate installed (Seafile Linux client recognizes it fine).

**Comment by bluscr33n** [comment]
I guess the problem are the supported SSL protocols / ciphers. By default Android 4.3 and below are not able to use TLS 1.2 or TLS 1.1 (see https://www.ssllabs.com/ssltest/clients.html). As @SpeedProg stated, enabling TLS 1.0 on server-side fixed the problem. However, you may don't want to enable TLS 1.0 or SSL3 and below due security reasons.

As far as I understand, it is possible to enable TLS 1.1/1.2 support on Android 4.1+ via:
```java
ssl.setEnabledProtocols(ssl.getSupportedProtocols());
```
(Should be adapt to enable only secure protocols, see http://blog.dev001.net/post/67082904181/android-using-sni-and-tlsv1-2-with-apache)

I'm facing the same problem in my setup - running a private seafile server, only supporting TLS1.1 & TLS1.2, and two android devices (running 4.4.4 and 2.3.7). On both devices I get the 'NetworkError' message when creating an account.

**Comment by shoeper** [comment]
Afaik a PR has been made few days ago disabling SSLv3 and enabling newer
TLS versions.
Am 09.01.2015 18:29 schrieb "Tom" <notifications@github.com>:

**Comment by SpeedProg** [comment]
Yeah at least the problem I had should be fixed by 7eb70ba0398c79791e0bd98e3270030992d94908 and a6c6923ddbb69f43380190f88d5a36c13e1f8f40 to some extend.
Having SNI would be awesome, too.

**Comment by bluscr33n** [comment]
I can confirm that the commits fixed the problem, too. Thanks for the hint :)

**Comment by mrjovanovic** [comment]
Sorry it took me so long. Can confirm that the problem is solved in version 1.4.2.
